datatree/tests/test_io.py

```diff
@@ -18,6 +18,23 @@ def test_to_netcdf(self, tmpdir):
         roundtrip_dt = open_datatree(filepath)
         assert_equal(original_dt, roundtrip_dt)
 
+    @requires_netCDF4
+    def test_netcdf_compression(self, tmpdir):
+        filepath = str(
+            tmpdir / "test.nc"
+        )  # casting to str avoids a pathlib bug in xarray
+        original_dt = create_test_datatree()
+
+        # add compression
+        comp = dict(zlib=True, complevel=9)
+        enc = {"/set1": {var: comp for var in original_dt["/set1"].ds.data_vars}}
```

Comment:
I've confirmed that these encoding values are routed all the way through to netCDF4's `createVariable` method (https://github.com/pydata/xarray/blob/4cae8d0ec04195291b2315b1f21d846c2bad61ff/xarray/backends/netCDF4_.py#L488-L501).

Reply:
Thanks @kmuehlbauer!